substrate/frame/assets/src/functions.rs

```diff
@@ -1045,11 +1059,22 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		freezer: T::AccountId,
 	) -> DispatchResult {
 		let mut d = Asset::<T, I>::get(&id).ok_or(Error::<T, I>::Unknown)?;
-		d.owner = owner;
-		d.admin = admin;
-		d.issuer = issuer;
-		d.freezer = freezer;
+		if d.status == AssetStatus::LiveAndNoPrivileges {
+			d.status = AssetStatus::Live
+		}
+		match d.try_set_team(&owner, &issuer, &admin, &freezer) {
+			Ok(()) => (),
+			Err(SetTeamError::AssetStatusLiveAndNoPrivileges) => log::error!(
```

Comment:
Yes ok, I added a new error variant: `InternalError` and abort in case of such internal error.

https://github.com/paritytech/polkadot-sdk/pull/4150/commits/6594d94178eb4f4d87e35224466c6badad18f938